fmn/core/cli.py

```diff
@@ -21,3 +26,31 @@ def cli(settings_file: str | None):
     """Fedora Messaging Notifications"""
     if settings_file:
         config.set_settings_file(settings_file)
+
+
+@cli.group()
+def cleanup():
+    """Cleanup operations."""
+
+
+@cleanup.command()
+@click.option("--days", type=int, default=31, help="Expire entries older than these many days")
+def generated_count(days):
+    """Expire old records of generated notification counts."""
+
+    async def _doit():
+        limit = datetime.now() - timedelta(days=days)
+        await init_async_model()
+        async with async_session_maker() as db:
+            result = await db.execute(
+                select(func.count(Generated.id)).filter(Generated.when < limit)
+            )
+            count = result.scalar()
+            if count > 0:
+                click.echo(f"Expiring {count} entries older than {limit}")
+                await db.execute(delete(Generated).filter(Generated.when < limit))
+                await db.commit()
```

Comment:
This probably isn’t important, but I guess we could DELETE and count the deleted rows in one go rather than making this two queries, something like (untested):

```
    result = await db.execute(delete(Generated).filter(Generated.when < limit).returning(Generated.id))
    await db.commit()
    if result.rowcount > 0:
        click.echo(f"Expired {result.rowcount} entries older than {limit}")
    else:
        click.echo(f"Nothing to clean up")
```

Reply:
True, but I kinda like the idea of logging things before they happen, I think it makes it easier to debug. If we were in a performance-critical codepath I would totally adopt your suggestion though :-)